### DISCUSSION TITLE
Faster Inference with assume_straight_pages=False

### DISCUSSION THREAD

**Discussion by dhcracchiolo** [mindee/doctr#1605] [General]
When setting assume_straight_pages to false there are a lot of slow cv2 operations that are slowing down inference tremendously.  This is currently a big bottle neck for my setup.  Just curious if there's any easy way to speed things up?

P.S. I'm already orienting my images straight however due to creases and such in the pages I'm still benefitting from rotated rectangular bounding boxes.

**Comment by felixdittrich92** [comment]
Hi @dhcracchiolo :wave:,

What hardware do you run on ? 

You can play around with `det_bs` and `reco_bs` to fit what your hardware can handle.

For CPU workloads you can also take a look at: https://github.com/felixdittrich92/OnnxTR

On GPU (PyTorch backend) you can move the whole ocr_predictor to `.cuda()` and use half precision `.cuda().half()`

Best regards,
Felix
